File: solutions/zig/03-ff0/code/src/main.zig
const std = @import("std");
var stdin = std.fs.File.stdin().readerStreaming(&.{});
var stdout = std.fs.File.stdout().writerStreaming(&.{});

pub fn main() !void {
    while (true) {
        std.debug.print("$ ", .{});

        var input_buffer: [1024]u8 = undefined;
        var input_len = try stdin.read(&input_buffer);
        if (input_len > 0) input_len -= 1; // Remove the newline character
        const input_slice = input_buffer[0..input_len];

        try stdout.interface.print("{s}: command not found\n", .{input_slice});
    }
}

File: solutions/typescript/03-ff0/code/app/main.ts
import { createInterface } from "readline";

const rl = createInterface({
  input: process.stdin,
  output: process.stdout,
  prompt: "$ ",
});

rl.prompt();

rl.on("line", (command) => {
  console.log(`${command}: command not found`);
  rl.prompt();
});
